projects/admin/src/app/record/custom-editor/document-editor/document-editor.component.ts

```diff
@@ -73,11 +102,14 @@ export class DocumentEditorComponent {
   loadingChanged(value: boolean) {
     if (value === false) {
       combineLatest([this._route.params, this._route.queryParams])
-        .subscribe(([params, queryParams]) => {
-          if (queryParams.source != null && queryParams.pid != null) {
-            this.importFromExternalSource(queryParams.source, queryParams.pid);
-          }
-        });
+      .subscribe(([params, queryParams]) => {
+        if (queryParams.source != null && queryParams.pid != null) {
```

Comment:
you can factorize: `queryParams.pid != null` in a separate `if` statement.

Reply:
Changed